planner/core/planbuilder.go

```diff
@@ -1463,6 +1463,9 @@ func (b *PlanBuilder) buildAdminCheckTable(ctx context.Context, as *ast.AdminStm
 	if !ok {
 		return nil, infoschema.ErrTableNotExists.GenWithStackByArgs(tblName.DBInfo.Name.O, tableInfo.Name.O)
 	}
+	if tableInfo.TempTableType == model.TempTableGlobal || tableInfo.TempTableType == model.TempTableLocal {
```

Comment:
Could we put this check to `preprocess.go`?  You can refer to ` checkCreateDatabaseGrammar `.